parts/k8s/addons/coredns.yaml

```diff
@@ -5,14 +5,14 @@ metadata:
   namespace: kube-system
   labels:
       kubernetes.io/cluster-service: "true"
-      addonmanager.kubernetes.io/mode: Reconcile
+      addonmanager.kubernetes.io/mode: EnsureExists
```

Comment:
In order to better anticipate user-configuration of coredns, this PR will update the addon mode to EnsureExists

Reply:
Hmm... This move also caused `coredns` to never be upgraded to newer configurations on a cluster upgrade. It also introduced #3478 due to the use of the `create` operation instead of the `apply`.

Reply:
A quick question here (I attempted to find out but couldn't identify the mechanism, sorry): 

Does the coredns configurations as part of the clusterAPI redeploy the `kube-dns` service if the pod IP changes or does it use addon-manager to deploy the change? If so, during a cluster upgrade, those changes wouldn't be propagated since the service is set to `EnsureExist`.